src/basic/os-util.c

```diff
@@ -345,6 +367,7 @@ int os_release_support_ended(const char *support_end, bool quiet, usec_t *ret_eo
                  * ourselves. */
 
                 r = parse_os_release(NULL,
+                                     IMAGE_EXTENSION,
```

Comment:
this is confusing, this is *not* checking an extension but parsing the regular *full* *OS* image. Hence this should probably pass _IMAGE_CLASS_INVALID here (or possibly IMAGE_MACHINE)